test/fw/ptl/utils/plugins/ptl_test_info.py

```diff
@@ -128,16 +129,22 @@ def wantClass(self, cls):
         """
         Is the class wanted?
         """
+        if not issubclass(cls, unittest.TestCase) or cls is unittest.TestCase:
```

Comment:
If I am not wrong then with this fix it will still report num of suites as 2 (given and parent one which is PBSTestSuite) and num of test cases as total num of tests in given suite (for SmokeTest it is ~50).
@ericpershey Can you please check the last two lines of `pbs_benchpress -t SmokeTest -L` output and see what it reports? it should report num of suites as 1 and num of test cases as ~50.

And if my theory is right and it above output reports num of suites > 1 then we need to add here `cls is PBSTestSuite` 
and in `finalize()` change `if self.list_test or self.gen_ts_tree:` to `if (self.list_test and not self.suites) or self.gen_ts_tree`.

Reply:
Total number of Test Suites: 2
Total number of Test Cases: 51

We could not figure out what it was really suppose to display.  Since it is two, I will go ahead and make the modifications you suggest.

Reply:
@ericpershey Thanks! and about -L options in pbs_benchpress, it should display the test cases in given suite (-t)/file (-f) with counts or if no suite/file is given then it should display the hierarchy of suites (aka subclasses of PBSTestSuite).

Reply:
Just checked it, here is what it displays which seems correct.  Both -L and -t SmokeTest -L are provided.

[vagrant@pbs-sn1 tests]$ pbs_benchpress -t SmokeTest -L


Test Suite: SmokeTest

    file: tests/pbs_smoketest.py

    module: tests.pbs_smoketest

    Tags: smoke

    Suite Doc: 
    
        This test suite contains a few smoke tests of PBS
    
        
    Test Cases: 
	test_submit_job
	test_submit_job_array
	test_advance_reservation
	test_standing_reservation
	test_degraded_advance_reservation
	test_select
	test_alter
	test_sigjob
	test_backfilling
	test_hold_release
	test_create_vnode
	test_create_execution_queue
	test_create_routing_queue
	test_fgc_limits
	test_limits
	test_finished_jobs
	test_project_based_limits
	test_job_scheduling_order
	test_preemption
	test_preemption_qrun
	test_fairshare
	test_server_hook
	test_mom_hook
	test_shrink_to_fit
	test_submit_job_with_script
	test_formula_match
	test_staging
	test_route_queue
	test_movejob
	test_by_queue
	test_round_robin
	test_pbs_probe
	test_printjob
	test_comm_service
	test_add_server_dyn_res
	test_schedlog_preempted_info
	test_basic
	test_basic_ja
	test_shrink_to_fit_resv_barrier
	test_job_sort_formula_threshold
	test_suspend_job_with_preempt
	test_resume_job_with_preempt
	test_suspend_job_array_with_preempt
	test_resume_job_array_with_preempt
	test_resource_create
	test_resource_delete
	test_fairshare_enhanced
	test_interactive_job
	test_man_pages
	test_exclhost
	test_jobscript_max_size


Total number of Test Suites: 1
Total number of Test Cases: 51
[vagrant@pbs-sn1 tests]$ 
[vagrant@pbs-sn1 tests]$ pbs_benchpress -L


Test Suite: PBSTestSuite

    file: ptl/utils/pbs_testsuite.py

    module: ptl.utils.pbs_testsuite

    Suite Doc: 
    
        Generic ``setup``, ``teardown``, and ``logging`` functions to
        be used as parent class for most tests.
        Class instantiates:
    
        ``server object connected to localhost``
    
        ``scheduler objected connected to localhost``
    
        ``mom object connected to localhost``
    
        Custom parameters:
    
        :param server: The hostname on which the PBS ``server/scheduler``
                       are running
        :param mom: The hostname on which the PBS MoM is running
        :param servers: Colon-separated list of hostnames hosting a PBS server.
                        Servers are then accessible as a dictionary in the
                        instance variable servers.
        :param client: For CLI mode only, name of the host on which the PBS
                       client commands are to be run from. Format is
                       ``<host>@<path-to-config-file>``
        :param moms: Colon-separated list of hostnames hosting a PBS MoM.
                     MoMs are made accessible as a dictionary in the instance
                     variable moms.
        :param comms: Colon-separated list of hostnames hosting a PBS Comm.
                      Comms are made accessible as a dictionary in the
                      instance variable comms.
        :param nomom=<host1>\:<host2>...: expect no MoM on given set of hosts
        :param mode: Sets mode of operation to PBS server. Can be either
                     ``'cli'`` or ``'api'``.Defaults to API behavior.
        :param conn_timeout: set a timeout in seconds after which a pbs_connect
                             IFL call is refreshed (i.e., disconnected)
        :param skip-setup: Bypasses setUp of PBSTestSuite (not custom ones)
        :param skip-teardown: Bypasses tearDown of PBSTestSuite (not custom ones)
        :param procinfo: Enables process monitoring thread, logged into
                         ptl_proc_info test metrics. The value can be set to
                         _all_ to monitor all PBS processes,including
                         ``pbs_server``, ``pbs_sched``, ``pbs_mom``, or a process
                         defined by name.
        :param revert-to-defaults=<True|False>: if False, will not revert to
                                                defaults.True by default.
        :param revert-hooks=<True|False>: if False, do not revert hooks to
                                          defaults.Defaults to True.
                                          ``revert-to-defaults`` set to False
                                          overrides this setting.
        :param del-hooks=<True|False>: If False, do not delete hooks. Defaults
                                       to False.``revert-to-defaults`` set to
                                       False overrides this setting.
        :param revert-queues=<True|False>: If False, do not revert queues to
                                           defaults.Defaults to True.
                                           ``revert-to-defaults`` set to False
                                           overrides this setting.
        :param revert-resources=<True|False>: If False, do not revert resources
                                              to defaults. Defaults to True.
                                              ``revert-to-defaults`` set to False
                                              overrides this setting.
        :param del-queues=<True|False>: If False, do not delete queues. Defaults
                                        to False.``revert-to-defaults`` set to
                                        Falseoverrides this setting.
        :param del-vnodes=<True|False>: If False, do not delete vnodes on MoM
                                        instances.Defaults to True.
        :param server-revert-to-defaults=<True|False>: if False, don't revert
                                                       Server to defaults
        :param comm-revert-to-defaults=<True|False>: if False, don't revert Comm
                                                     to defaults
        :param mom-revert-to-defaults=<True|False>: if False, don't revert MoM
                                                    to defaults
        :param sched-revert-to-defaults=<True|False>: if False, don't revert
                                                      Scheduler to defaults
        :param procmon: Enables process monitoring. Multiple values must be
                        colon separated. For example to monitor ``server``,
                        ``sched``, and ``mom`` use
                        ``procmon=pbs_server:pbs_sched:pbs_mom``
        :param procmon-freq: Sets a polling frequency for the process monitoring
                             tool.Defaults to 10 seconds.
        :param test-users: colon-separated list of users to use as test users.
                           The users specified override the default users in the
                           order in which they appear in the ``PBS_USERS`` list.
        :param default-testcase-timeout: Default test case timeout value.
        :param data-users: colon-separated list of data users.
        :param oper-users: colon-separated list of operator users.
        :param mgr-users: colon-separated list of manager users.
        :param root-users: colon-separated list of root users.
        :param build-users: colon-separated list of build users.
        :param clienthost: the hostnames to set in the MoM config file
        

    Test suite hierarchy:
        TestFunctional
            Test_Rrecord_with_resources_used
            TestAcctLog
            TestPbsAccumulateRescUsed
            Test_acl_groups
            Test_acl_host_moms
            TestAdminSuspend
            TestSchedAllPart
            TestAlpsInventoryCheckHook
            TestCrayAlpsReleaseTunables
            Test_array_job_email
            TestBasilParserErrors
            TestCalendaring
            TestCgroupsHook
            TestCheckpoint
            TestClientResponse
            Test_complete_running_parent_job
            TestResvStaleVnode
            TestPBSConfig
            TestPbsCpuset
            TestCpusetDestroyDelay
            TestCheckNodeExclusivity
            TestCrayHyperthread
            TestCrayPaggIdUniqueness
            TestPbsReliableJobStartupOnCray
            TestCraySmokeTest
            TestSuspendResumeOnCray
            TestVnodePerNumaNode
            TestVnodePool
            TestDupAccLogForResv
            TestEligibleTime
            TestEquivClass
            TestPbsExecjobSuspendResume
            TestFairshare
            TestHighResLogging
            TestHolidays
            TestHookConfigOSEnv
            TestPbsHookCrossLinkMom
            TestHookDebugInput
            TestHookDebugNoCrash
            TestHookExechostPeriodic
            TestPbsExecjobAbort
            TestPbsExecjobEnd
            TestPbsExecutePrologue
            Test_hook_perf_stat
            TestPbsHookSetJobEnv
            TestHookSetNonExist
            TestHookTimeout
            TestHookUnsetRes
            TestHookSmokeTest
            TestHookSwig
            TestHostResources
            TestPbsInitScript
            TestJobArray
            TestJobArrayComment
            TestJobComment
            TestJobPurge
            TestJobRequeueTimeoutErrorMsg
            TestJobRouting
            TestPbsJobScript
            TestJobSortFormula
            Test_job_status_after_mom_hup
            TestJobTask
            TestMaintenanceReservations
            TestMomDynRes
            TestMomHookSync
            TestMomJobDir
            TestMomWalltime
            TestMovedJob
            TestMultipleSchedulers
            TestExecJobHook
            TestNodeBuckets
            TestNodeJobsRestart
            TestPbsNodeRampDown
            TestPbsnodes_json
            TestNodesQueues
            TestNonprintingCharacters
            TestOfflineVnode
            Test_single_event_multiple_hooks
            Test_explicit_psets
            TestOnlySmallFilesOverTPP
            Test_passing_environment_variable_via_qsub
            TestPbsnodes
            TestPbsnodesOutputTrimmed
            Test_periodic_constant
            Test_power_provisioning_cray
            Test_power_provisioning_sgi
            TestPreemption
            TestProvisioningJob
            TestProvisioningJob_Enh
            TestPythonRestartSettings
            Test_pbs_python
            TestQmgr
            TestQrun
            TestQselect
            TestQstat
            TestQstatTwoServers
            TestqstatStateCount
            TestQstatFormats
            TestQsub_direct_write
            TestQsubOptionsArguments
            TestQsub_remove_files
            TestPbsResvAlter
            TestReleaseLimitedResOnSuspend
            TestPbsReliableJobStartup
            Test_Custom_Resource_Perm
            Test_singleNode_Job_ResourceUsed
            TestReservations
            TestResourceMultiChunk
            TestResourceUsageLog
            TestResvEndHook
            TestReservationRequests
            Test_RootOwnedScript
            TestPbsRstat
            TestRunJobHook
            TestSchedFifo
            TestSchedLoadBalancing
            TestSchedPreemptEnforceResumption
            TestSchedSubjobBadstate
            TestIndirectResources
            Test_server_periodic_hook
            TestMomEnforcement
            TestSisterMom
            TestPBSSnapshot
            TestSoftWalltime
            TestSTF
            TestStrictOrderingAndBackfilling
            TestSupportLinuxHookEventPhase1_2
            TestServerDynRes
            Test_systemd
            TestEntityLimits
            Test_qorder
            Test_run_count
            TestServerDefaultAttrib
            TestTrillionJobid
            TestAcctlogRescUsedWithTwoMomHooks
            Test_pbs_types
            TestUnknownResourceHookUpdate
            Testunset_exectime
            TestQsubWithQueuejobHook
            TestVerifyLogOutput
        classcomp
        TestDeleteJobUsingServerTaggedInJobId
        TestInterfaces
            TestNodePartition
            TestPartition
            TestPreemptParamsQmgr
            TestSchedulerInterface
        TestPerformance
            TestCgroupsStress
            TestClientNagles
            TestJobEquivClassPerf
            HistoryCleanupQuasihang
            TestJobPerf
            TestPreemptPerformance
            TestQstatPerformance
            TestQsubPerformance
            JobRerunFileTransferPerf
            TestSchedPerf
            StandingResvQuasihang
        TestResilience
            TestPbsHookAlarmLargeMultinodeJob
        TestSecurity
        TestSelf
            PBSTestCycle
            TestDefaultTimeout
            TestDshUtils
            TestExpect
            TestPbsInitServices
            TestJobCleanup
            TestJSONReport
            TestLogMatch
            TestManager
            TestManagersOperators
            TestPBSTestSuite
            TestRequirementsDecorator
            TestPTLConvertsResvIDtoid
            Test_create_vnodes
            TestRevertSiteHook
            TestPTLRevertToDefault
            Test_TestparamsDecorator
        TestUpgrades
        SmokeTest
        TestFunctional
            Test_Rrecord_with_resources_used
            TestAcctLog
            TestPbsAccumulateRescUsed
            Test_acl_groups
            Test_acl_host_moms
            TestAdminSuspend
            TestSchedAllPart
            TestAlpsInventoryCheckHook
            TestCrayAlpsReleaseTunables
            Test_array_job_email
            TestBasilParserErrors
            TestCalendaring
            TestCgroupsHook
            TestCheckpoint
            TestClientResponse
            Test_complete_running_parent_job
            TestResvStaleVnode
            TestPBSConfig
            TestPbsCpuset
            TestCpusetDestroyDelay
            TestCheckNodeExclusivity
            TestCrayHyperthread
            TestCrayPaggIdUniqueness
            TestPbsReliableJobStartupOnCray
            TestCraySmokeTest
            TestSuspendResumeOnCray
            TestVnodePerNumaNode
            TestVnodePool
            TestDupAccLogForResv
            TestEligibleTime
            TestEquivClass
            TestPbsExecjobSuspendResume
            TestFairshare
            TestHighResLogging
            TestHolidays
            TestHookConfigOSEnv
            TestPbsHookCrossLinkMom
            TestHookDebugInput
            TestHookDebugNoCrash
            TestHookExechostPeriodic
            TestPbsExecjobAbort
            TestPbsExecjobEnd
            TestPbsExecutePrologue
            TestHookManagement
            Test_hook_perf_stat
            TestPbsHookSetJobEnv
            TestHookSetNonExist
            TestHookTimeout
            TestHookUnsetRes
            TestHookSmokeTest
            TestHookSwig
            TestHostResources
            TestPbsInitScript
            TestJobArray
            TestJobArrayComment
            TestJobComment
            TestJobPurge
            TestJobRequeueTimeoutErrorMsg
            TestJobRouting
            TestPbsJobScript
            TestJobSortFormula
            Test_job_status_after_mom_hup
            TestJobTask
            TestMaintenanceReservations
            TestMomDynRes
            TestMomHookSync
            TestMomJobDir
            TestMomWalltime
            TestMovedJob
            TestMultipleSchedulers
            TestExecJobHook
            TestNodeBuckets
            TestNodeJobsRestart
            TestPbsNodeRampDown
            TestPbsnodes_json
            TestNodesQueues
            TestNonprintingCharacters
            TestOfflineVnode
            Test_single_event_multiple_hooks
            Test_explicit_psets
            TestOnlySmallFilesOverTPP
            Test_passing_environment_variable_via_qsub
            TestPbsnodes
            TestPbsnodesOutputTrimmed
            Test_periodic_constant
            Test_power_provisioning_cray
            Test_power_provisioning_sgi
            TestPreemption
            TestProvisioningJob
            TestProvisioningJob_Enh
            TestPythonRestartSettings
            Test_pbs_python
            TestQmgr
            TestQrun
            TestQselect
            TestQstat
            TestQstatTwoServers
            TestqstatStateCount
            TestQstatFormats
            TestQsub_direct_write
            TestQsubOptionsArguments
            TestQsub_remove_files
            TestPbsResvAlter
            TestReleaseLimitedResOnSuspend
            TestPbsReliableJobStartup
            Test_Custom_Resource_Perm
            Test_singleNode_Job_ResourceUsed
            TestReservations
            TestResourceMultiChunk
            TestResourceUsageLog
            TestResvEndHook
            TestReservationRequests
            Test_RootOwnedScript
            TestPbsRstat
            TestRunJobHook
            TestSchedFifo
            TestSchedLoadBalancing
            TestSchedPreemptEnforceResumption
            TestSchedSubjobBadstate
            TestIndirectResources
            Test_server_periodic_hook
            TestMomEnforcement
            TestSisterMom
            TestPBSSnapshot
            TestSoftWalltime
            TestSTF
            TestStrictOrderingAndBackfilling
            TestSupportLinuxHookEventPhase1_2
            TestServerDynRes
            Test_systemd
            TestEntityLimits
            Test_qorder
            Test_run_count
            TestServerDefaultAttrib
            TestTrillionJobid
            TestAcctlogRescUsedWithTwoMomHooks
            Test_pbs_types
            TestUnknownResourceHookUpdate
            Testunset_exectime
            TestQsubWithQueuejobHook
            TestVerifyLogOutput
        classcomp
        TestDeleteJobUsingServerTaggedInJobId
        TestInterfaces
            TestNodePartition
            TestPartition
            TestPreemptParamsQmgr
            TestSchedulerInterface
        TestPerformance
            TestCgroupsStress
            TestClientNagles
            TestJobEquivClassPerf
            HistoryCleanupQuasihang
            TestJobPerf
            TestPreemptPerformance
            TestQstatPerformance
            TestQsubPerformance
            JobRerunFileTransferPerf
            TestSchedPerf
            StandingResvQuasihang
        TestResilience
            TestPbsHookAlarmLargeMultinodeJob
        TestSecurity
        TestSelf
            PBSTestCycle
            TestDefaultTimeout
            TestDshUtils
            TestExpect
            TestPbsInitServices
            TestJobCleanup
            TestJSONReport
            TestLogMatch
            TestManager
            TestManagersOperators
            TestPBSTestSuite
            TestRequirementsDecorator
            TestPTLConvertsResvIDtoid
            Test_create_vnodes
            TestRevertSiteHook
            TestPTLRevertToDefault
            Test_TestparamsDecorator
        TestUpgrades
        SmokeTest


Total number of Test Suites: 1
Total number of Test Cases: 0
[vagrant@pbs-sn1 tests]$